sdc/datatypes/common_functions.py

```diff
@@ -639,3 +640,55 @@ def _sdc_pandas_series_check_axis_impl(axis):
         return _sdc_pandas_series_check_axis_impl
 
     return None
+
+
+def _sdc_pandas_series_align(series, other, size='max', finiteness=False):
+    """
+    Align series and other series by
+        size where size of output series is max/min size of input series
+        finiteness where all the infinite and matched finite values are replaced with nans, e.g.
+        series: [1., inf, inf, -1.,   0.] -> [1., nan, nan, -1.,   0.]
+        other:  [1., -1.,  0., 0.1, -0.1] -> [1., nan, nan, 0.1, -0.1]
+    """
+    pass
+
+
+@sdc_overload(_sdc_pandas_series_align, jit_options={'parallel': False})
+def _sdc_pandas_series_align_overload(series, other, size='max', finiteness=False):
+    ty_checker = TypeChecker('Function sdc.common_functions._sdc_pandas_series_align().')
+    ty_checker.check(series, SeriesType)
+    ty_checker.check(other, SeriesType)
+
+    str_types = (str, types.StringLiteral, types.UnicodeType, types.Omitted)
+    if not isinstance(size, str_types):
+        ty_checker.raise_exc(size, 'str', 'size')
+
+    if not isinstance(finiteness, (bool, types.Boolean, types.Omitted)):
+        ty_checker.raise_exc(finiteness, 'bool', 'finiteness')
+
+    def _sdc_pandas_series_align_impl(series, other, size='max', finiteness=False):
+        if size != 'max' and size != 'min':
+            raise ValueError("Function sdc.common_functions._sdc_pandas_series_align(). "
+                             "The object size\n expected: 'max' or 'min'")
+
+        arr, other_arr = series._data, other._data
+        arr_len, other_arr_len = len(arr), len(other_arr)
+        min_length = min(arr_len, other_arr_len)
+        length = max(arr_len, other_arr_len) if size == 'max' else min_length
+
+        aligned_arr = numpy.array([numpy.nan] * length)
```

Comment:
You could probably use numpy.repeat(numpy.nan, length), but I doubt it's parallelized. So it might turn out that using numpy.array is faster.

Reply:
Measured `res = numpy.array([numpy.nan] * 10 ** 8)` vs. `numpy.repeat(numpy.nan, 10 ** 8)` with `SDC_AUTO_PARALLEL = True`

name | nthreads | type | size | median | min | max | compile | boxing
-- | -- | -- | -- | -- | -- | -- | -- | --
numpy.array | 1.0 | SDC | 100000000.0 | 1.0090000629425049 | 1.002000093460083 | 1.022000074386597 | 0.09800267219543456 | 0.09000849723815918
numpy.array | 4.0 | SDC | 100000000.0 | 1.2319998741149902 | 1.0130000114440918 | 2.2649998664855957 | 0.0879981517791748 | 0.09700202941894531
numpy.repeat | 1.0 | SDC | 100000000.0 | 0.4199998378753662 | 0.3970000743865967 | 0.49299979209899897 | 0.15896368026733398 | 5.0067901611328125e-06
numpy.repeat | 4.0 | SDC | 100000000.0 | 0.3919999599456787 | 0.3859999179840088 | 0.4030001163482666 | 0.15500760078430176 | 1.6689300537109375e-06

I don't see scalability at all. `numpy.repeat` looks faster than `numpy.array` in this case. So let me replace `numpy.array` with `numpy.repeat`.

Reply:
I measured performance based on SDC performance test system:
https://github.com/IntelPython/sdc/compare/master...densmirn:perf/array_vs_repeat